base/number.jl

```diff
@@ -68,7 +68,9 @@ ndims(x::Number) = 0
 ndims(::Type{<:Number}) = 0
 length(x::Number) = 1
 firstindex(x::Number) = 1
+firstindex(x::Number, ::Int) = 1
```

Comment:
Since `size` and `axes` do this check as well, this should probably be:
```suggestion
firstindex(x::Number, d::Int) = d < 1 ? throw(BoundsError()) : 1
```